src/dataset.ts

```diff
@@ -15,15 +15,16 @@
  */
 
 import {DecorateRequestOptions, DeleteCallback, ServiceObject} from '@google-cloud/common';
-import {paginator} from '@google-cloud/paginator';
+import {paginator, ResourceStream} from '@google-cloud/paginator';
 import {promisifyAll} from '@google-cloud/promisify';
 import * as extend from 'extend';
 import * as r from 'request';
 import {Duplex, Readable} from 'stream';
 import {teenyRequest} from 'teeny-request';
 
-import {BigQuery, DatasetCallback, Query, QueryRowsResponse, SimpleQueryRowsCallback} from '.';
+import {BigQuery, DatasetCallback, PagedCallback, PagedRequest, PagedResponse, Query, QueryRowsResponse, ResourceCallback, SimpleQueryRowsCallback} from '.';
 import {JobCallback, JobResponse, Table, TableMetadata, TableOptions} from './table';
+import {bigquery} from './types';
```

Comment:
If I'm understanding correctly, we'd already solved this bug in `googleapis`, but were not using the same type definitions in this project?

Reply:
Yeah, my best guess was whoever implemented the types initially just looked to see what options we explicitly referenced and as a result they were largely incomplete.

Reply:
Hey, that was me!  And yea, they ain't great.

Reply:
Commence the public shaming!


Just kidding, I ❤️ you